x/leverage/keeper/reserves.go

```diff
@@ -46,6 +46,25 @@ func (k Keeper) setReserveAmount(ctx sdk.Context, coin sdk.Coin) error {
 	return nil
 }
 
+// checkBadDebt detects if a borrower has zero non-blacklisted collateral,
+// and marks any remaining borrowed tokens as bad debt.
+func (k Keeper) checkBadDebt(ctx sdk.Context, borrowerAddr sdk.AccAddress) error {
+	// get remaining collateral uTokens, ignoring blacklisted
+	remainingCollateral := k.filterAcceptedUTokens(ctx, k.GetBorrowerCollateral(ctx, borrowerAddr))
+
+	// detect bad debt if collateral is completely exhausted
+	if remainingCollateral.IsZero() {
```

Comment:
what if the liquidator paid everything and took all collateral? Then the position is closed and there is no collateral and no bad debt.

Reply:
Correct, if collateral is zero and there are no nonzero borrows, no bad debt records are created.

Reply:
But in this case, the function will mark bad debt anyway.

Reply:
`for _, coin := range k.GetBorrowerBorrows(ctx, borrowerAddr)`

This is an empty `sdk.Coins` if no borrows remain, so the body of the for loop is never executed.

Reply:
That's right.